torch/csrc/distributed/autograd/context/dist_autograd_container.h

```diff
@@ -113,9 +132,16 @@ class TORCH_API DistAutogradContainer {
   // and worker_id_ are immutable.
   mutable std::mutex autograd_context_lock_;
 
+  // Atomic var to control when the watchdog thread is killed.
+  std::atomic<bool> terminateWatchdog_;
+
   // Autograd message id to identify unique send/recv autograd function pairs.
   std::atomic<int64_t> next_autograd_message_id_;
 
+  // Set Timeout after which watchdog thread will clean up
+  // DistAutogradContexts.
+  std::chrono::duration<int64_t> cleanupContextTimeout;
```

Comment:
@pritamdamania87 has previously suggested in other diffs that timeouts be millisecond level granularity when we store them, so that they are more granular, though I'm not sure if this applies in this diff. Initialization like `std::chrono::milliseconds kContextCleanupTimeout = std::chrono::seconds(600)` will still work and it'll automatically convert. One benefit to this is that it can make the python test even faster.